syntax = "proto3";

package ingestion;

option go_package = ".;proto_gen";



// -----------------------------------------------------------------------------
// Internal Services
// -----------------------------------------------------------------------------

// InternalAuthService provides a private API for the ingestion service to
// validate API keys and notify about new data.
service InternalAuthService {
  // ValidateApiKey checks if an API key is valid.
  rpc ValidateApiKey(ValidateApiKeyRequest) returns (ValidateApiKeyResponse) {}

  // NotifyNewParquetFile tells the backend to index a newly flushed parquet file.
  // Called by ingestion after a cold flush completes.
  rpc NotifyNewParquetFile(NotifyNewParquetFileRequest) returns (NotifyNewParquetFileResponse) {}
}

message ValidateApiKeyRequest {
  // The API key to be validated.
  string api_key = 1;
}

message ValidateApiKeyResponse {
  // Whether the API key is valid.
  bool is_valid = 1;
}

message NotifyNewParquetFileRequest {
  // Absolute path to the newly created parquet file.
  string file_path = 1;
}

message NotifyNewParquetFileResponse {
  // Whether the file was successfully indexed.
  bool indexed = 1;
  // Number of spans indexed from the file.
  int64 span_count = 2;
}
